x/observer/genesis.go

```diff
@@ -94,21 +94,30 @@ func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState)
 		k.SetLastObserverCount(ctx, &types.LastObserverCount{LastChangeHeight: 0, Count: observerCount})
 	}
 
+	tss := types.TSS{}
 	if genState.Tss != nil {
-		tss := *genState.Tss
+		tss = *genState.Tss
 		k.SetTSS(ctx, tss)
+	}
+
+	// Set all the pending nonces
+	if genState.PendingNonces != nil {
+		for _, pendingNonce := range genState.PendingNonces {
+			k.SetPendingNonces(ctx, pendingNonce)
+		}
+	} else {
```

Comment:
maybe move condition `if genState.Tss != nil` to this line to make it a `else if genState.Tss != nil {` ?